### DISCUSSION TITLE
adding bitwise logging to console.sol

### DISCUSSION THREAD

**Discussion by CryptoKiddies** [NomicFoundation/hardhat#4377] [Show and tell]
Hi team, I had this utility created to see the bitwise values in a word for testing and debugging contract logic, particularly when using bit masks. Wondering if you'd be interested in adding support directly in console.sol with this util? I'm suggesting a feature indicated with a `%b` flag perhaps. Cheers! 

code: https://github.com/YOLOrekt/community-bug-bounty/blob/bounty-head/contracts/utils/LogBinary.sol

**Comment by fvictorio** [comment]
Hi @CryptoKiddies, thanks for sharing this. The way `console.sol` works right now is that the arguments are pretty much forwarded to [`util.format`](https://nodejs.org/api/util.html#utilformatformat-args), which sadly doesn't support binary numbers.

That being said, I think this is a reasonable feature request. I wouldn't do it at the solidity level though. I think we should pre-process the arguments and handle things like `%b` (we already do things similar to this, so it wouldn't be terribly weird). I can't promise we'll be able to implement it soon, but seems like a super reasonable feature request, if you want to open an issue for it.

**Comment by CryptoKiddies** [comment]
A bit of a delay here... yes, definitely think this would be useful for low level developers (which are still commonplace for now) in the solidity ecosystem. Any kind of bitwise operations or state packing optimization testing would benefit from being able to see the word printed out as 1s and 0s. Wonder if I can contribute, but definitely agree this is pre-processed.
